vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -182,6 +189,9 @@ type Config struct {
 	// If not specify any in flags, then genericapiserver will only enable defaultAPIResourceConfig.
 	MergedResourceConfig *serverstore.ResourceConfig
 
+	// LifecycleEventSink receives events about the life cycle of the API server, e.g. readiness, serving, signals and termination.
+	LifecycleEventSink record.EventSink
```

Comment:
do we need this public?

Reply:
yes, the kube-apiserver has to override it.